src/Controls/src/Core/Compatibility/Handlers/ListView/iOS/ListViewRenderer.cs

```diff
@@ -690,6 +690,8 @@ void ReloadData()
 				Control.ReloadData();
 			else
 				PerformWithoutAnimation(() => { Control.ReloadData(); });
+				
+			Control.ScrollToRow(NSIndexPath.FromRowSection(0, 0), UITableViewScrollPosition.Bottom, false);
```

Comment:
This changes are impacting some tests:

<img width="334" alt="image" src="https://github.com/user-attachments/assets/37e94620-9b74-4faf-ade3-fef75c91c50a">

Crashing:
`ObjCRuntime.ObjCException : Objective-C exception thrown.  Name: NSRangeException Reason: Attempted to scroll the table view to an out-of-bounds row (0) when there are only 0 rows in section 0. Table view: <UITableView: 0x1136f8200; frame = (0 0; 375 2); clipsToBounds = YES; autoresize = W+H; gestureRecognizers = <NSArray: 0x600000f94a50>; backgroundColor = <UIDynamicSystemColor: 0x6000017c1f40; name = tableBackgroundColor>; layer = <CALayer: 0x6000006b9460>; contentOffset: {0, 0}; contentSize: {375, 28}; adjustedContentInset: {0, 0, 0, 0}; dataSource: <Microsoft_Maui_Controls_Handlers_Compatibility_ListViewRenderer_UnevenListViewDataSource: 0x6000041f5070>>\nNative stack trace:\n\t0   CoreFoundation                      0x00000001804b70ec __exceptionPreprocess + 172\n\t1   libobjc.A.dylib                     0x000000018008ede8 objc_exception_throw + 72\n\t2   CoreFoundation                      0x00000001804b6ffc -[NSException initWithCoder:] + 0\n\t3   UIKitCore                           0x0000000185cda8b4 -[UITableView _contentOffsetForScrollingToRowAtIndexPath:atScrollPosition:usingPresentationValues:] + 360\n\t4   UIKitCore                           0x0000000185cdb338 -[UITableView _scrollToRowAtIndexPath:atScrollPosition:animated:usingPresentationValues:] + 136\n\t5   UIKitCore                           0x0000000185cdb1e0 -[UITableView scrollToRowAtIndexPath:atScrollPosition:animated:] + 136\n\t6   libxamarin-dotnet-debug.dylib       0x0000000103588a14 xamarin_dyn_objc_msgSend + 160\n\t7   Microsoft.Maui.Controls.DeviceTests 0x0000000101a8d89c interp_to_native_trampoline + 172\n\t8   libmonosgen-2.0.dylib               0x00000001039a8584 ves_pinvoke_method + 584\n\t9   libmonosgen-2.0.dylib               0x000000010399c5f4 mono_interp_exec_method + 3072\n\t10  libmonosgen-2.0.dylib               0x0000000103999f60 interp_runtime_invoke + 244\n\t11  libmonosgen-2.0.dylib               0x00000001038ad1e0 mono_jit_runtime_invoke + 1064\n\t12  libmonosgen-2.0.dylib               0x0000000103a7d700 mono_runtime_try_invoke + 156\n\t13  libmonosgen-2.0.dylib               0x0000000103a80774 mono_runtime_invoke + 488\n\t14  Microsoft.Maui.Controls.DeviceTests 0x0000000101aaf7c4 _ZL30native_to_managed_trampoline_9P11objc_objectP13objc_selectorPP11_MonoMethodj + 316\n\t15  Microsoft.Maui.Controls.DeviceTests 0x0000000101ab00e4 -[__MonoMac_NSAsyncSynchronizationContextDispatcher xamarinApplySelector] + 44\n\t16  Foundation                          0x0000000180f520d8 __NSThreadPerformPerform + 124\n\t17  CoreFoundation                      0x000000018041b324 CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION + 24\n\t18  CoreFoundation                      0x000000018041b26c __CFRunLoopDoSource0 + 172\n\t19  CoreFoundation                      0x000000018041a9d0 __CFRunLoopDoSources0 + 232\n\t20  CoreFoundation                      0x00000001804150b0 __CFRunLoopRun + 788\n\t21  CoreFoundation                      0x0000000180414960 CFRunLoopRunSpecific + 536\n\t22  GraphicsServices                    0x0000000190183b10 GSEventRunModal + 160\n\t23  UIKitCore                           0x0000000185aa2b40 -[UIApplication _run] + 796\n\t24  UIKitCore                           0x0000000185aa6d38 UIApplicationMain + 124\n\t25  libxamarin-dotnet-debug.dylib       0x00000001035471e4 xamarin_UIApplicationMain + 60\n\t26  libmonosgen-2.0.dylib               0x00000001039a99ec do_icall + 316\n\t27  libmonosgen-2.0.dylib               0x00000001039a81e8 do_icall_wrapper + 356\n\t28  libmonosgen-2.0.dylib               0x000000010399c43c mono_interp_exec_method + 2632\n\t29  libmonosgen-2.0.dylib               0x0000000103999f60 interp_runtime_invoke + 244\n\t30  libmonosgen-2.0.dylib               0x00000001038ad1e0 mono_jit_runtime_invoke + 1064\n\t31  libmonosgen-2.0.dylib               0x0000000103a7c6ec mono_runtime_invoke_checked + 148\n\t32  libmonosgen-2.0.dylib               0x0000000103a83e88 mono_runtime_exec_main_checked + 116\`

Reply:
It will work now :)